src/services/coordinateformat.js

```diff
@@ -0,0 +1,178 @@
+goog.provide('ngeo.CoordinateFormat');
+goog.provide('ngeo.CoordinateFormatConfig');
+
+goog.require('goog.math');
+goog.require('goog.string');
+goog.require('ol.coordinate');
+goog.require('ol.proj');
+goog.require('ngeo');
+
+
+/**
+ * @typedef {{
+ *   code: (string|undefined),
+ *   id: (string),
+ *   label: (string),
+ *   format: (function(ol.Coordinate))
+ * }}
+ */
+ngeo.CoordinateFormatConfig;
+
+/**
+ * The `ngeoCoordinateFormat` provides a list of projection config & formaters
+ * that are used to display coordinates.
+ * The service provides a default config list in `projectionFormat`. This
+ * list is public and can be extended. You can pick an item from the list
+ * with `getProjection` function and overrides it.
+ * To get a subset of this array, you can use `getProjections` function.
+ *
+ * @constructor
+ * @export
+ * @ngInject
+ * @ngdoc service
+ * @ngname gmfMousepositionController
+ */
+ngeo.CoordinateFormat = function() {
+
+  if (typeof proj4 == 'function') {
+
+    proj4.defs('EPSG:32631', '+proj=utm +zone=31 +ellps=WGS84 ' +
+        '+datum=WGS84 +units=m +no_defs ');
+
+    proj4.defs('EPSG:32632', '+proj=utm +zone=32 +ellps=WGS84 ' +
+        '+datum=WGS84 +units=m +no_defs ');
+  }
+
+  /**
+   * @type {Array.<ngeo.CoordinateFormatConfig>}
+   * @export
+   */
+  this.projectionFormat = [{
+    id: 'EPSG:2056',
+    code: 'EPSG:2056',
+    label: 'CH1903+ / LV95',
+    format: coordinatesFormat_
+  },{
+    id: 'EPSG:21781',
+    label: 'CH1903 / LV03',
+    format: coordinatesFormat_
+  },{
+    id: 'EPSG:4326',
+    label: 'Lon/Lat WGS84',
+    format: function(coordinates) {
+      return ol.coordinate.format(coordinates, ' {x} E | {y} N', 5);
+    }
+  },{
+    id: 'EPSG:4326:DMS',
+    code: 'EPSG:4326',
+    label: 'Lon/Lat WGS84 DMS',
+    format: function(coordinates) {
+      var hdms = toStringHDMS_(coordinates);
+      var yhdms = hdms.split(' ').slice(0, 4).join(' ');
+      var xhdms = hdms.split(' ').slice(4, 8).join(' ');
+      return xhdms + ' | ' + yhdms;
+    }
+  },{
+    id: 'EPSG:4326:UTM3132',
+    code: 'EPSG:4326',
+    label: 'WGS84 UTM 32|31',
+    format: function(coordinates) {
+      if (coordinates[0] < 6 && coordinates[0] >= 0) {
+        var utm_31t = ol.proj.transform(coordinates,
+            'EPSG:4326', 'EPSG:32631');
+        return ol.coordinate.format(utm_31t, '{x} | {y} (UTM32N)', 0);
+      } else if (coordinates[0] < 12 && coordinates[0] >= 6) {
+        var utm_32t = ol.proj.transform(coordinates,
+            'EPSG:4326', 'EPSG:32632');
+        return ol.coordinate.format(utm_32t, '{x} | {y} (UTM32N)', 0);
+      } else {
+        return '-';
+      }
+    }
+  }];
```

Comment:
> This s so custom, should be in the application controller ...

Why not, it was just to suggest some default values, as it is over writable.

> and the "projections" EPSG:4326:DMS, EPSG:4326:UTM3132 looks like an hack ...

It is the way it's done in geoadmin & lux